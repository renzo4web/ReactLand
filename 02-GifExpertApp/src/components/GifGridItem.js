import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ url, title }) => {
  return (
    <li className="hover:z-10">
      <img
        src={url}
        alt={title}
        className="transform transition-transform ease-in rounded-md hover:scale-125 "
      />
    </li>
  );
};
GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
export default GifGridItem;
