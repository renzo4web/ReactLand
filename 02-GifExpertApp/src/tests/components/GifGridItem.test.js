import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("GifGridItem test", () => {
  const url = "https://www.example.com";
  const title = "title";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("should render a gif", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should img have the attr from src=url alt=title", () => {
    const { alt, src } = wrapper.find("img").props();
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });
});
