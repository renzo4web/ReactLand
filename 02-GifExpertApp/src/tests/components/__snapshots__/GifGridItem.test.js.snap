// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem test should render a gif 1`] = `
<li
  className="hover:z-10"
>
  <img
    alt="title"
    className="transform transition-transform ease-in rounded-md hover:scale-125 "
    src="https://www.example.com"
  />
</li>
`;
